protocol/mining/src/rules/blue_parents_only_rule.rs

```diff
@@ -0,0 +1,66 @@
+use std::sync::{
+    atomic::{AtomicBool, AtomicU64, Ordering},
+    Arc,
+};
+
+use kaspa_consensus_core::api::counters::ProcessingCountersSnapshot;
+use kaspa_core::{trace, warn};
+
+use super::{mining_rule::MiningRule, ExtraData};
+
+const VIRTUAL_PROCESSING_TRIGGER_THRESHOLD: f64 = 500.0; // 500 milliseconds
+const VIRTUAL_PROCESSING_RECOVERY_THRESHOLD: f64 = 100.0; // 100 milliseconds
+
+/// BlueParentsOnlyRule
+/// Attempt to recover from high virtual processing times possibly caused merging red blocks.
+/// by only pointing to blue parents.
+///
+/// Trigger: virtual processing average time is above threshold
```

Comment:
I think slowness in `build_block_template` should trigger it. @hashdag and I also suggested slowness in header validation to trigger it